lib/srv/usermgmt.go

```diff
@@ -218,6 +218,10 @@ func (u *HostSudoersManagement) WriteSudoers(name string, sudoers []string) erro
 		sudoersOut.WriteString(fmt.Sprintf("%s %s\n", name, entry))
 	}
 	err := u.backend.WriteSudoersFile(name, []byte(sudoersOut.String()))
+	if errors.Is(err, host.ErrInvalidSudoers) {
+		log.Warnf("encountered invalid sudoers entry in: %s", sudoersOut.String())
```

Comment:
Use structured logging whenever possible:
```suggestion
		log.WithField("sudoers_entry", sudoersOut.String()).Warn("Encountered invalid sudoers entry.")
```

Should we truncate the string? Do we have concerns about it being potentially sensitive?

Reply:
> Should we truncate the string? Do we have concerns about it being potentially sensitive?

I wondered about this, but I couldn't think of a good reason not to off the top of my head 🤔 An alternative, that actually might be more useful to end-users, would be to validate the entries while resolving matching rules. That way we could inform them which role has the broken entry instead of falling back to logging the entire sudoers file and leaving it up to them to figure out which role needs attention.

Reply:
I rebased this on the slog changes and adjusted the message we're logging a bit. The log and the error are a bit redundant in cases where the returned error also ends up getting logged (e.g. in `regular/sshserver.go`), but I think that's probably okay since this shouldn't happen unless there's an incorrectly configured resource. And I think it's better in this case to ensure we're always surfacing something actionable when this happens